docs/adding_packages/build_and_package.md

```diff
@@ -0,0 +1,36 @@
+# Build and Package
+
+This document gathers all the relevant information regarding the general lines to follow while writing either the `build()` or the `package()` methods.
+Both of these methods often take advantage of build helpers build binaries and install to the `package_folder`.
+
+<!-- toc -->
+## Contents
+
+  * [Build Method](#build-method)
+  * [Package](#package)<!-- endToc -->
+
+## Build Method
+
+* `build()` method should focus on build only, not installation. During the build, nothing should be written in `package` folder. Installation step should only occur in `package()` method.
+
+* The build itself should only rely on local files. Nothing should be downloaded from internet during this step. If external files are required, they should come from `requirements` or `build_requirements`, in addition to source files downloaded in `source()` or coming from recipe itself.
+
+* Except CMake and a working build toolchain (compiler, linker, archiver etc), the recipe should not assume that any other build tool is installed on user build machine (like Meson, autotools or pkg-config). On Windows, recipe should not assume that a shell is available (like MSYS2). Therefore, if the build requires additional tools, they should be added to `build_requirements()`.
+
+* It is forbidden to run other conan client commands during build. In other words, if upstream build files call conan under the hood (through `cmake-conan` for example or any other logic), this logic must be removed.
+
+* Settings from profile should be honored (`build_type`, `compiler.libcxx`, `compiler.cppstd`, `compiler.runtime` etc).
+
+* These env vars from profile should be honored and properly propagated to underlying build system during the build: `CC`, `CXX`, `CFLAGS`, `CXXFLAGS`, `LDFLAGS`.
```

Comment:
it deserves an example. for newcomers, it's kinda unclear. can you illustrate how to honor these variables, e.g. for CMake?

Reply:
Yes the docs are bad - I am not trying to fix them in this PR 😄 I got many more too come!

Reply:
I suppose we need to avoid confusion and use consistent naming/terminology across the docs. either `tool_requires` everywhere, or `build_requires` everywhere.